packages/ra-ui-materialui/src/button/SaveButton.tsx

```diff
@@ -62,100 +59,56 @@ import { FormRenderProps } from 'react-final-form';
 export const SaveButton = (props: SaveButtonProps) => {
     const {
         className,
+        icon = defaultIcon,
         invalid,
         label = 'ra.action.save',
-        disabled,
-        redirect,
-        saving,
-        submitOnEnter,
-        variant = 'contained',
-        icon = defaultIcon,
         onClick,
-        handleSubmitWithRedirect,
-        onSave,
-        onSuccess,
         onFailure,
+        onSuccess,
+        saving,
+        disabled = saving,
+        submitOnEnter,
         transform,
+        variant = 'contained',
         ...rest
     } = props;
-
-    const notify = useNotify();
     const translate = useTranslate();
-    const formContext = useFormContext();
-    const { setOnSuccess, setOnFailure, setTransform } = useSaveContext(props);
-
-    const handleClick = event => {
-        // deprecated: use onSuccess and transform instead of onSave
-        if (typeof onSave === 'function') {
```

Comment:
This should be documented in the upgrade guide

Reply:
Done